### ISSUE TITLE
Run maskrcnn onnx from pytorch and inference on c++ with gpu sometimes will error

### ISSUE THREAD

**Issue by LuckGuySam** [microsoft/onnxruntime#10543]
**Question**
I use pytorch official maskrcnn export onnx model, and run the model on c++ library with gpu sometimes will error.
![image](https://user-images.githubusercontent.com/74532070/153810578-22e0a1b9-955c-4581-894b-bca42604b1c1.png)

run on c++with cpu it never happen, and i find the node on netron. It only Loop_299 exsit, Concat_2443 can't find

![image](https://user-images.githubusercontent.com/74532070/153810623-d7d8b70d-ea58-40a9-bd62-2dd43bd82a47.png)

the result I always use the same image as input, and I run the loop just with inference

![image](https://user-images.githubusercontent.com/74532070/153810658-fd485da1-aca3-469b-a992-92ec24469b51.png)

other test: I run the model in python on gpu is no error
I am not sure that is it a bug or someone can help me to resolve?

**env**
python : 3.7.11
pytorch : 1.9.0
torchvision : 0.10.0
onnxruntime : 1.8.1
cuda : 11.1
cudnn : 8.2.1
c++ : VC 2017

**Comment by gwang-msft** [comment]
`Concat_2443` may exist in the subgraph of node `Loop_299`, care to share the model?

**Comment by LuckGuySam** [comment]
this is my model 
https://drive.google.com/file/d/1jvrTYt3b0SvdQCOElYa5-9yAVvFB8wZp/view?usp=sharing

also can use this code to get model

    import torch
    import numpy as np
    import onnxruntime as rt
    
    model = torchvision.models.detection.maskrcnn_resnet50_fpn(pretrained=True)
    model.eval()
    x = [torch.rand(3, 300, 500)]
    predictions = model(x)
    input_names = ["image"]
    output_names = ['boxes', 'labels', 'scores', 'masks']
    torch.onnx.export(model,
     x,
     "mask_rcnn.onnx",
     input_names=input_names,
     output_names=output_names,
     dynamic_axes={
        "boxes":[0],
        "labels":[0],
        "scores":[0],
        "masks":[0,2,3],
        
     },
     opset_version=13)

After testing, this phenomenon will occur in opset 11 to 13

**Comment by suneesh** [comment]
I am facing exactly the same problem. Any lead on this will be highly appreciated.
